packages/daemon/src/runner/helper.ts

```diff
@@ -15,19 +15,24 @@ import {
 
 const { PLUGINS, PIPCOOK_LOGS } = constants;
 
-export async function parseConfig(configPath: string, generateId = true): Promise<PipelineDB> {
+export async function parseConfig(configPath: string | RunConfigI, generateId = true): Promise<PipelineDB> {
   let configJson: RunConfigI;
-  const urlObj = url.parse(configPath);
-  if (urlObj.protocol === null) {
-    throw new TypeError('config URI is not supported');
-  }
-  if ([ 'http:', 'https:' ].indexOf(urlObj.protocol) >= 0) {
-    configJson = JSON.parse(await request(configPath));
-  } else if (urlObj.protocol === 'file:') {
-    configJson = await fs.readJSON(url.fileURLToPath(configPath));
+  if (typeof configPath === 'string') {
+    const urlObj = url.parse(configPath);
+    if (urlObj.protocol === null) {
+      throw new TypeError('config URI is not supported');
+    }
+    if ([ 'http:', 'https:' ].indexOf(urlObj.protocol) >= 0) {
+      configJson = JSON.parse(await request(configPath));
+    } else if (urlObj.protocol === 'file:') {
+      configJson = await fs.readJSON(url.fileURLToPath(configPath));
+    } else {
+      throw new TypeError(`protocol ${urlObj.protocol} is not supported`);
+    }
   } else {
-    throw new TypeError(`protocol ${urlObj.protocol} is not supported`);
+    configJson = configPath;
   }
```

Comment:
Shall we have a new function called `loadConfig`?

Reply:
Yes, I modified and define a new func